tests/v2/validation/workloads/workload_test.go

```diff
@@ -316,38 +349,109 @@ func (w *WorkloadTestSuite) TestWorkloadPodScaleDown() {
+	containerName := namegen.AppendRandomString("pauserediscontainer")
+	newContainerTemplate := workloads.NewContainer(containerName,
+		redisImageName,
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	pauseDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	log.Info("Updating deployment image and replica")
+	pauseDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, pauseDeployment)
+	require.NoError(w.T(), err)
+
+	log.Info("Verifying deployment was not update")
+	log.Infof("Counting all pods running by image %s", imageName)
+	countPods, err := pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, imageName)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+
+	log.Info("Activing orchestration")
+	pauseDeployment.Spec.Paused = false
+	pauseDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, pauseDeployment)
+
+	w.validateDeploymentScale(namespace.Name, pauseDeployment, redisImageName, 2)
+
+	log.Info("Verifying orchestration is active")
+	err = deployment.VerifyOrchestrationStatus(w.client, w.cluster.ID, namespace.Name, pauseDeployment.Name, false)
+	require.NoError(w.T(), err)
+
+	log.Infof("Counting all pods running by image %s", redisImageName)
+	countPods, err = pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, redisImageName)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+}
+
 func (w *WorkloadTestSuite) validateDeploymentUpgrade(namespaceName string, appv1Deployment *appv1.Deployment, expectedRevision string, image string, expectedReplicas int) {
+	log.Info("Waiting deployment comes up active")
 	err := charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespaceName, metav1.ListOptions{
 		FieldSelector: "metadata.name=" + appv1Deployment.Name,
 	})
 	require.NoError(w.T(), err)
 
+	log.Info("Waiting for all pods to be running")
 	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespaceName, appv1Deployment)
 	require.NoError(w.T(), err)
 
-	err = deployment.ValidateRolloutHistoryDeployment(w.client, w.cluster.ID, namespaceName, appv1Deployment.Name, expectedRevision)
+	log.Infof("Verifying rollout history by revision %s", expectedRevision)
+	err = deployment.VerifyRolloutHistoryDeployment(w.client, w.cluster.ID, namespaceName, appv1Deployment.Name, expectedRevision)
 	require.NoError(w.T(), err)
 
+	log.Infof("Counting all pods running by image %s", image)
 	countPods, err := pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespaceName, image)
 	require.NoError(w.T(), err)
 	require.Equal(w.T(), expectedReplicas, countPods)
 }
 
 func (w *WorkloadTestSuite) validateDeploymentScale(namespaceName string, scaleDeployment *appv1.Deployment, image string, expectedReplicas int) {
+	log.Info("Waiting deployment comes up active")
```

Comment:
since you're already editing this, can you move the helpers out to a new file in this folder called something like `workload.go`?

Reply:
Yes, I can

Reply:
ping'd offline to clarify